crates/ef-testing/src/models/case.rs

```diff
@@ -177,11 +173,14 @@ impl BlockchainTestCase {
             let starknet_contract_address =
                 StarknetContractAddress(Into::<StarkFelt>::into(starknet_address).try_into()?);
 
-            let actual_state = starknet.storage.get(&starknet_contract_address);
+            let starknet_db = starknet
+                .maybe_as_cached_db()
+                .ok_or_else(|| RunnerError::Other("failed to get Katana database".to_string()))?;
+            let actual_state = starknet_db.storage.get(&starknet_contract_address);
             match actual_state {
                 None => {
                     // if no state, check post state is empty
-                    let actual_balance = read_balance(evm_address, starknet_address, &starknet)
+                    let actual_balance = read_balance(evm_address, starknet_address, &mut starknet)
```

Comment:
why does read_balance need a mutable reference to starknet?

Reply:
I was also triggered by this but it comes from the fact that read_balance uses the `Database` from Katana in order to read, and the Database is a super-trait for the `StateReader` from `blockifier`, which takes a mutable reference even for reading.
https://github.com/dojoengine/dojo/blob/main/crates/katana/core/src/fork/db.rs#L100